spec/spec_helper.rb

```diff
@@ -1,116 +1,74 @@
-require 'simplecov'
-SimpleCov.start
+require 'codeclimate-test-reporter'
+SimpleCov.start do
+  add_filter '/spec/'
 
+  formatter SimpleCov::Formatter::MultiFormatter[
+    SimpleCov::Formatter::HTMLFormatter,
+    CodeClimate::TestReporter::Formatter
+  ]
+end
+
+require 'dotenv'
+Dotenv.load
+
+require 'blockscore'
+require 'vcr'
 require 'factory_girl'
 require 'webmock'
 require 'webmock/rspec'
 require 'rspec'
 require 'rspec/its'
+require 'pry'
 
 BlockScore::Spec.setup
 
-# This file was generated by the `rspec --init` command. Conventionally, all
-# specs live under a `spec` directory, which RSpec adds to the `$LOAD_PATH`.
-# The generated `.rspec` file contains `--require spec_helper` which will cause
-# this file to always be loaded, without a need to explicitly require it in any
-# files.
-#
-# Given that it is always loaded, you are encouraged to keep this file as
-# light-weight as possible. Requiring heavyweight dependencies from this file
-# will add to the boot time of your test suite on EVERY test run, even for an
-# individual file that may not need all of that loaded. Instead, consider making
-# a separate helper file that requires the additional dependencies and performs
-# the additional setup, and require it from the spec files that actually need
-# it.
-#
-# The `.rspec` file also contains a few flags that are not defaults but that
-# users commonly want.
-#
-# See http://rubydoc.info/gems/rspec-core/RSpec/Core/Configuration
-
 module BlockScore
+  VCR.configure do |c|
+    c.cassette_library_dir = 'spec/cassettes'
+    c.hook_into :webmock
+    c.configure_rspec_metadata!
+    c.ignore_hosts 'codeclimate.com'
+
+    c.filter_sensitive_data('BLOCKSCORE_TEST_KEY') { ENV['BLOCKSCORE_TEST_KEY'] }
+  end
+
   RSpec.configure do |config|
     config.include(FactoryGirl::Syntax::Methods)
-    config.include(WebMock::API)
 
-    config.before(:each) do
-      @api_stub = stub_request(:any, Spec::STUB_PATTERN)
-                  .with(headers: Spec::HEADERS)
-                  .to_return(&Spec.method(:webmock_handler))
+    config.before(:suite) do
+      BlockScore.api_key = ENV['BLOCKSCORE_TEST_KEY'] || 'BLOCKSCORE_TEST_KEY'
```

Comment:
Its not an error. Its the matching filler text so that if you don't have a key, your request matches the URI sorted in the sanitized cassette. The tests run with or without a value key.